yargs/index.d.ts

```diff
@@ -36,6 +36,12 @@ declare namespace yargs {
         demand(positionals: number, msg: string): Argv;
         demand(positionals: number, max: number, msg?: string): Argv;
```

Comment:
It is still technically valid code - would a [`@deprecated`](http://usejsdoc.org/tags-deprecated.html) doc suffice here?